bin/middlewares/map_style.js

```diff
@@ -0,0 +1,51 @@
+const styleBuilder = require('@qwant/map-style-builder')
+const styleConfigure = require('mapbox_style_configure')
+const qwantStyle = require('@qwant/qwant-basic-gl-style/style.json')
+const path = require('path')
+
+
+module.exports = function (config, languages) {
+
+  /* pre-build style on server start */
+  let options = {
+    output: 'production', // 'debug' | 'production' | 'omt'
+    outPath : __dirname + '/../public/mapstyle',
+    i18n : true,
+    icons : true,
+    pixelRatios : [1,2],
+    styleDir : path.resolve(path.join(__dirname, '..', '..', 'node_modules', '@qwant' , 'qwant-basic-gl-style')),
+    conf : {tileserver_base : '{tileserver_base}', tileserver_poi : '{tileserver_poi}', spriteserver : '{spriteserver}', fontserver : '{fontserver}'},
+    ignoreSprite : true
+  }
+
+  let builtStyle = styleBuilder(qwantStyle, options)
+
+  return function (req, res) {
+    /* get lang routine */
+    let langParameter = req.query.lang
+    let matchedLanguage = languages.defaultLanguage
+    if(langParameter) {
+      matchedLanguage = languages.supportedLanguages.find((supportedLanguage) => supportedLanguage.code === langParameter)
+      if(!matchedLanguage) {
+        res.status(400)
+        res.send({error : `Unsupported language code : ${langParameter}`})
+        return
+      }
+    } else {
+      matchedLanguage = languages.defaultLanguage
+    }
+
+    /* config rewriting */
+    let host = req.get('host')
+    let protocol = req.protocol
```

Comment:
`req.protocol` will use the `X-Forwarded-Proto` header to determine protocol.  
But it seems that in our docker swarm environments these header will be overridden with `http` by internal services. 
So it is not reliable, and we should probably always use `https` here.

Reply:
hmmm ok. i'm impressed

Reply:
it will be harder to test on local by replacing style object by url

Reply:
we could work around that, and use `http` when `host === 'localhost'` ?
Or define a new setting with default `https` that we could override in local environments ?

Reply:
:S i prefer not and say it's exclusive for external securised services